src/momento/internal/synchronous/_scs_grpc_manager.py

```diff
@@ -46,15 +49,75 @@ class _DataGrpcManager:
     version = momento_version
 
     def __init__(self, configuration: Configuration, credential_provider: CredentialProvider):
+        self._logger = logs.logger
         self._secure_channel = grpc.secure_channel(
             target=credential_provider.cache_endpoint,
             credentials=grpc.ssl_channel_credentials(),
         )
+
         intercept_channel = grpc.intercept_channel(
             self._secure_channel, *_interceptors(credential_provider.auth_token, configuration.get_retry_strategy())
         )
         self._stub = cache_client.ScsStub(intercept_channel)  # type: ignore[no-untyped-call]
 
+        self._eagerly_connect(configuration)
+
+    """
+        This method tries to connect to Momento's server eagerly in async fashion until
+        EAGER_CONNECTION_TIMEOUT elapses.
+    """
+
+    def _eagerly_connect(self, configuration: Configuration) -> None:
+        eager_connection_timeout: timedelta = (
+            configuration.get_transport_strategy().get_grpc_configuration().get_eager_connection_timeout()
+        )
+
+        # An event to track whether we were able to establish an eager connection
+        connection_event = threading.Event()
```

Comment:
We need this so that our `subscription` on line 115 doesn't lurk around even after it's job is done